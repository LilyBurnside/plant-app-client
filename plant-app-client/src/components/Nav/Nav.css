.Nav {
  height: 40px;
  display: flex;
  align-items: normal;
  justify-content: space-between;
  background-color: #EFF9F4;
}

.Nav h1 {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  margin-left: 10px;
}

.Nav a {
  color: #07312F;
  text-decoration: none;
}


.Nav__tagline--wide {
  display: none;
}

.Nav__tagline--narrow {
  opacity: 0.35;
  display: block;
  text-align: center;
  display: none;
  margin-top: 10px;
}

.Nav__logged-in {
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: flex-end;
}

.Nav__not-logged-in {
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: flex-end;
}

.Nav__logged-in a:last-child {
  background: #D58BFA;
  color: white;
  padding: 5px 15px;
  border-radius: 15px;
  margin-left: 10px;
  margin-right: 10px;
}

.Nav__not-logged-in a {
  background: #D58BFA;
  color: white;
  padding: 5px 15px;
  border-radius: 15px;
  margin-left: 10px;
  margin-right: 10px;
}

@media screen and (min-width: 750px) {
  /* .Nav {
    align-items: baseline;
  } */

  .Nav h1 {
    font-size: 22pt;
  }

  .Nav__tagline--wide {
    opacity: 0.35;
    margin-left: 10px;
  }
  
}

/* @media screen and (max-width: 490px) {
  
} */
