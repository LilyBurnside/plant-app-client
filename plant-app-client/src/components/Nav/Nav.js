import React, { Component } from 'react'
import { Link } from 'react-router-dom'
//import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import TokenService from '../../services/token-service'
import './Nav.css'

export default class Nav extends Component {
  handleLogoutClick = () => {
    TokenService.clearAuthToken()
  }

  renderLogoutLink() {
    return (
      <div className='Nav__logged-in'>
        <Link
          onClick={this.handleLogoutClick}
          to='/'>
          Logout
        </Link>
        <Link
          to='/wishlist'>
            My wishlist
        </Link>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='Nav__not-logged-in'>
        <Link
          to='/login'>
          Log in
        </Link>
      </div>
    )
  }

  render() {
    return <>
      <nav className='Nav'>
        <h1>
          <Link to='/'>
            {/* <FontAwesomeIcon className='blue' icon='gift' /> */}
            {' '}
            Plant Shelf
          </Link>
        </h1>
        <span className='Nav__tagline--wide'>Find your plant</span>
        {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
      </nav>

     
    </>
  }
}