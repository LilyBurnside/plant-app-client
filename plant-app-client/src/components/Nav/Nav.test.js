import React from 'react';
import ReactDOM from 'react-dom';
// import { shallow } from 'enzyme'
// import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom';
import Nav from './Nav';

describe('Nav Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });


});