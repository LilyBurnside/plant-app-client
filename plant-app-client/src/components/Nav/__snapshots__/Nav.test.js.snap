// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component renders UI as expected if not logged in 1`] = `
<Fragment>
  <nav
    className="Nav"
  >
    <h1>
      <Link
        to="/"
      >
         
        Plant Shelf
      </Link>
    </h1>
    <span
      className="Nav__tagline--wide"
    >
      Find your plant
    </span>
    <div
      className="Nav__not-logged-in"
    >
      <Link
        to="/login"
      >
        Log in
      </Link>
    </div>
  </nav>
</Fragment>
`;
